### ISSUE TITLE
Not dropping other groups when doing setuid

### ISSUE THREAD

**Issue by xambroz** [friedrich/hans#15]
Hello.
Not dropping other groups might leave the new process with some unwanted groups inherited. 
I would recommend adding following patch:

```
diff -ru hans-0.4.4.orig/src/worker.cpp hans-0.4.4.new/src/worker.cpp
--- hans-0.4.4.orig/src/worker.cpp      2014-02-07 04:36:31.000000000 +0100
+++ hans-0.4.4.new/src/worker.cpp       2016-11-17 16:12:12.313698244 +0100
@@ -200,6 +200,9 @@

     syslog(LOG_INFO, "dropping privileges");

+    if (setgroups(0, NULL) == -1)
+        throw Exception("setgroups", true);
+
     if (setgid(gid) == -1)
         throw Exception("setgid", true);
```

**Comment by friedrich** [comment]
Thank you for noticing! Fixed in ed3d7de6cc76a39a26b85df59db595c4746b61e0.

**Comment by xambroz** [comment]
Cool .. thx.
